platform/windows/detect.py

```diff
@@ -178,8 +178,15 @@ def configure_msvc(env, manual_msvc_config):
     """Configure env to work with MSVC"""
 
     # Build type
+
     if env["tests"]:
         env["windows_subsystem"] = "console"
+    elif env["windows_subsystem"] == "default":
+        # Default means we use console for debug, gui for release.
+        if "debug" in env["target"]:
```

Comment:
This make it impossible to build tools in GUI mode since it can't be in "release" mode.
Do we want it for "debug" and "release_debug" or just for "debug"?

Reply:
`3.2` uses console for both `debug` and `release_debug`, which I think make sense. `release_debug` is used for debug export templates, and most users who use those likely want to have debug output. Similarly, nightly builds use `release_debug` for the editor, and it's confusing currently that you can't get console output when testing those builds.

Might be worth reconsidering for the stable branch though and make it `gui` for `release_debug`, as the main incentive for this change was people being bothered by their editor having a console I guess.

I really wish #41164 could have worked. Windows' subsystem all or nothing system is really a pain.